tracer/src/Datadog.Trace/ClrProfiler/ServerlessInstrumentation/AWS/LambdaCommon.cs

```diff
@@ -100,9 +93,14 @@ internal static bool SendStartInvocation(ILambdaExtensionRequest requestBuilder,
             return ValidateOKStatus((HttpWebResponse)request.GetResponse());
         }
 
-        internal static bool SendEndInvocation(ILambdaExtensionRequest requestBuilder, bool isError)
+        internal static bool SendEndInvocation(ILambdaExtensionRequest requestBuilder, bool isError, string data)
         {
             var request = requestBuilder.GetEndInvocationRequest(isError);
+            var byteArray = Encoding.UTF8.GetBytes(data ?? DefaultJson);
+            request.ContentLength = byteArray.Length;
+            Stream dataStream = request.GetRequestStream();
+            dataStream.Write(byteArray, 0, byteArray.Length);
+            dataStream.Close();
             return ValidateOKStatus((HttpWebResponse)request.GetResponse());
```

Comment:
nit-all that block could be factorize as it's a duplicate of `SendStartInvocation` body